sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/resource/ecs.py

```diff
@@ -0,0 +1,81 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import socket
+
+from opentelemetry.sdk.resources import Resource, ResourceDetector
+from opentelemetry.semconv.resource import (
+    CloudPlatformValues,
+    CloudProviderValues,
+    ResourceAttributes,
+)
+
+logger = logging.getLogger(__name__)
+
+_CONTAINER_ID_LENGTH = 64
+
+
+class AwsEcsResourceDetector(ResourceDetector):
+    """Detects attribute values only available when the app is running on AWS
+    Elastic Container Service (ECS) and returns them in a Resource.
+    """
+
+    def detect(self) -> "Resource":
+        try:
+            if not os.environ.get(
+                "ECS_CONTAINER_METADATA_URI"
+            ) and not os.environ.get("ECS_CONTAINER_METADATA_URI_V4"):
+                is_on_ecs_e_msg = "Missing ECS_CONTAINER_METADATA_URI therefore process is not on ECS."
+                logger.warning(
```

Comment:
Logging message would be logged twice in this case?

Reply:
Hm yes it would be you're right... I'll remove this one since the message in the `RunTimeError()` should make it should up in the later `logger.warning` message. Thanks!